packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -212,8 +212,10 @@ class HotRunner extends ResidentRunner {
             await device.generator!.compileExpression(expression, definitions,
                 definitionTypes, typeDefinitions, typeBounds, typeDefaults,
                 libraryUri, klass, method, isStatic);
-        if (compilerOutput != null && compilerOutput.expressionData != null) {
+        if (compilerOutput != null && compilerOutput.errorCount == 0 && compilerOutput.expressionData != null) {
           return base64.encode(compilerOutput.expressionData!);
+        } else if (compilerOutput!.errorCount > 0 && compilerOutput.errorMessage != null) {
```

Comment:
Don't you need a null check here, like on line 215?

```
if (compilerOutput != null && compilerOutput.errorCount == 0 && compilerOutput.expressionData != null) {
  return base64.encode(compilerOutput.expressionData!);
} else if (compilerOutput != null && compilerOutput.errorCount > 0 && compilerOutput.errorMessage != null) {
  throw VmServiceExpressionCompilationException(compilerOutput.errorMessage!);
}
```

Reply:
Good catch! Done.